server/polar/license_key/endpoints.py

```diff
@@ -0,0 +1,153 @@
+from fastapi import Depends
+from pydantic import UUID4
+
+from polar.authz.service import AccessType, Authz
+from polar.exceptions import ResourceNotFound, Unauthorized
+from polar.kit.db.postgres import AsyncSession
+from polar.kit.pagination import ListResource, PaginationParamsQuery
+from polar.models import LicenseKey, LicenseKeyActivation
+from polar.openapi import APITag
+from polar.organization.service import organization as organization_service
+from polar.postgres import get_db_session
+from polar.routing import APIRouter
+
+from . import auth
+from .schemas import (
+    LicenseKeyActivationRead,
+    LicenseKeyRead,
+    LicenseKeyUpdate,
+    LicenseKeyWithActivations,
+    NotFoundResponse,
+    UnauthorizedResponse,
+)
+from .service import license_key as license_key_service
+
+router = APIRouter(prefix="/license-keys", tags=[APITag.documented, APITag.featured])
+
+###############################################################################
+# LICENSE KEYS
+###############################################################################
+
+
+@router.get(
+    "",
+    response_model=ListResource[LicenseKeyRead],
+    responses={
+        401: UnauthorizedResponse,
+        404: NotFoundResponse,
+    },
+)
+async def list(
+    auth_subject: auth.LicenseKeysRead,
+    pagination: PaginationParamsQuery,
+    organization_id: UUID4,
```

Comment:
Lately, I tend to move away from mandatory/single-valued query parameter. I find it more elegant to make everything optional: by default, it'll show you all license keys you have access to (i.e. across all your organizations). That's the purpose of my `_get_readable_xx_statement` methods:

https://github.com/polarsource/polar/blob/ce29527fa5d323871e289f8bf3690dd2dafb3b81/server/polar/order/service.py#L499-L501

Then, to filter on one or several organization, you can use the `MultipleQueryFilter` helper:

https://github.com/polarsource/polar/blob/ce29527fa5d323871e289f8bf3690dd2dafb3b81/server/polar/order/endpoints.py#L34-L36

And then, extend the statement with:

https://github.com/polarsource/polar/blob/ce29527fa5d323871e289f8bf3690dd2dafb3b81/server/polar/order/service.py#L144-L145

---

The benefit of this approach is that:

* Users can finely tune the result of the query (no or multiple organization filter)
* You don't need to pre-check the existence of the organization, nor the access to it. If it doesn't exist or if you don't have permission, then you don't have results.

---

Anyway, we can deploy it like this and change it later: it won't be a breaking change to the API.

Reply:
Great callout & agree that the approach you've moved over to should be the convention and is better. Updating the implementation accordingly.